### ISSUE TITLE
NetDataWriter & NetDataReader Endianness issue

### ISSUE THREAD

**Issue by Azengar** [RevenantX/LiteNetLib#497]
Hi,

I noticed that the `NetDataWriter` and `NetDataReader` are using the `BitConverter` to write/read binary data.
Wouldn't it be better to use the `BinaryPrimitives` class to only write/read **Big Endian** binary data.

As far as I understand from MS documentation, the `BitConverter` writes using the OS endianness so it could mean that two machines with different endianness fail to communicate.

Thanks.

**Library version**: 0.9.5.2

**Framework**: .NET 6

**OS**: Windows

**Comment by RevenantX** [comment]
@Azengar hi! Which platforms that support .net use another endianness?

**Comment by Azengar** [comment]
Thanks for your quick answer.

As far as I understand ARM cpus can do both, so it would be possible for a modern Mac or a Linux machine to run in big endian mode.
I have no idea if that's common though, I'm not an expert on the matter.

**Comment by RevenantX** [comment]
@Azengar yes but they run on macs in "default" endianness. Only platform with different endianness if i remember correctly that was supported by old .net was Intel IA64

**Comment by Azengar** [comment]
I see, well if you think it's safe to assume that most if not all platforms are and will be little endian in the future then I guess I'll concede the point. :)
